langchain/src/retrievers/metal.ts

```diff
@@ -0,0 +1,36 @@
+import type MetalSDK from "@getmetal/metal-sdk";
+
+import { BaseRetriever } from "../schema/index.js";
+import { Document } from "../document.js";
+
+export interface MetalRetrieverFields {
+  client: MetalSDK.default;
+}
+
+interface ResponseItem {
+  text: string;
+  [key: string]: unknown;
+}
+
+export class MetalRetriever extends BaseRetriever {
+  client: MetalSDK.default;
+
+  constructor(fields: MetalRetrieverFields) {
+    super();
+
+    this.client = fields.client;
+  }
+
+  async getRelevantDocuments(query: string): Promise<Document[]> {
+    const res = await this.client.search({ text: query });
+
+    const items = ("data" in res ? res.data : res) as ResponseItem[];
+    return items.map(
+      ({ text, metadata }) =>
```

Comment:
we also return a `dist` which shows the similarity distance when running a search with a `text` query. Perhaps we should include this in the `metadata` of `Document` below?